core/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -193,24 +147,24 @@ public void dispatchRequest(final RestRequest request, final RestChannel channel
                     threadContext.putHeader(key, httpHeader);
                 }
             }
-            if (filters.length == 0) {
-                executeHandler(request, channel, client);
+
+            final RestHandler rawHandler = getHandler(request);
+            final RestHandler handler = handlerWrapper.apply(rawHandler);
+
+            if (handler != null) {
```

Comment:
sorry I dont' follow. if the wrapper returns null we say we didn't find a handler, are we ok with this?